Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1094 @@
+
+	bool isEmptySpace = false
+	int spaceType = 0
+	
+	// SLIDER BULLSHIT
+	bool sliderEnabled = false
+	float min = 0.0
+	float max = 1.0
+	float stepSize = 0.05
+	bool forceClamp = false
+
+	bool isCustomButton = false
+	void functionref() onPress
+
+	array<string> values
+	var customMenu
+	bool hasCustomMenu = false
+}
+
+struct {
+	var menu
+	int scrollOffset = 0
+	bool updatingList = false
+
+	array<ConVarData> conVarList
+	// if people use searches - i hate them but it'll do :)
+	array<ConVarData> filteredList
+	string filterText = ""
+	table< int, int > enumRealValues
+	table<string, bool> setFuncs
+	array<var> modPanels
+	array<MS_Slider> sliders
+	table settingsTable
+	string currentMod = ""
+	string currentCat = ""
+} file
+
+struct {
+	int deltaX = 0
+	int deltaY = 0
+} mouseDeltaBuffer
+
+void function AddModSettingsMenu()
+{
+	AddMenu( "ModSettings", $"resource/ui/menus/mod_settings.menu", InitModMenu )
+}
+
+void function InitModMenu()
+{
+	file.menu = GetMenu( "ModSettings" )
+	//DumpStack(2)
+	AddMenuFooterOption( file.menu, BUTTON_B, "#B_BUTTON_BACK", "#BACK" )
+
+	// Safe I/O stuff
+	// uncomment when safe i/o is merged.
+
+	/*
+	try
+	{
+		file.settingsTable = expect table( compilestring( "return NSLoadFile( \"Mod Settings\", \"settings\" )" )() )
+	}
+	catch (ex)
+	{
+	}
+
+	foreach (string key, var value in file.settingsTable)
+	{
+		printt(key, expect string( value ))
+		try
+		{
+			SetConVarString(key, expect string( value ))
+		}
+		catch (ex)
+		{
+			
+		}
+	}*/
+
+	/////////////////////////////
+	// BASE NORTHSTAR SETTINGS //
+	/////////////////////////////
+
+	// most of these are overrided in the cfg, maybe ask bob to remove the cfg stuff from there?
+	// at the same time, might fuck with dedis so idk.
+	// these are pretty long too, might need to e x t e n d the settings menu
+	AddModTitle( "#NORTHSTAR_BASE_SETTINGS" )
+	AddModCategory( "#PRIVATE_MATCH")
+	AddConVarSettingEnum("ns_private_match_only_host_can_change_settings", "#ONLY_HOST_MATCH_SETTINGS", [ "#NO", "#YES" ])
+	AddConVarSettingEnum("ns_private_match_only_host_can_change_settings", "#ONLY_HOST_CAN_START_MATCH", [ "#NO", "#YES" ])
+	AddConVarSettingSlider("ns_private_match_countdown_length", "#MATCH_COUNTDOWN_LENGTH", 0, 30, 0.5)
+	// probably shouldn't add this as a setting?
+	// AddConVarSettingEnum("ns_private_match_override_maxplayers", "Override Max Player Count", "Northstar - Server", [ "#NO", "#YES" ])
+	AddModCategory("#SERVER")
+	AddConVarSettingEnum("ns_should_log_unknown_clientcommands", "#LOG_UNKNOWN_CLIENTCOMMANDS", [ "#NO", "#YES" ])
+	AddConVarSetting("ns_disallowed_tacticals", "#DISALLOWED_TACTICALS")
+	AddConVarSetting("ns_disallowed_tactical_replacement", "#TACTICAL_REPLACEMENT")
+	AddConVarSetting("ns_disallowed_weapons", "#DISALLOWED_WEAPONS")
+	AddConVarSetting("ns_disallowed_weapon_primary_replacement", "#REPLACEMENT_WEAPON")
+	AddConVarSettingEnum("ns_should_return_to_lobby", "#SHOULD_RETURN_TO_LOBBY", [ "#NO", "#YES" ])
```

Comment:
```suggestion
	AddConVarSettingEnum( "ns_private_match_only_host_can_change_settings", "#ONLY_HOST_MATCH_SETTINGS", [ "#NO", "#YES" ] )
	AddConVarSettingEnum( "ns_private_match_only_host_can_change_settings", "#ONLY_HOST_CAN_START_MATCH", [ "#NO", "#YES" ])
	AddConVarSettingSlider("ns_private_match_countdown_length", "#MATCH_COUNTDOWN_LENGTH", 0, 30, 0.5)

	AddModCategory("#SERVER")
	AddConVarSettingEnum( "ns_should_log_unknown_clientcommands", "#LOG_UNKNOWN_CLIENTCOMMANDS", [ "#NO", "#YES" ] )
	AddConVarSetting( "ns_disallowed_tacticals", "#DISALLOWED_TACTICALS" )
	AddConVarSetting( "ns_disallowed_tactical_replacement", "#TACTICAL_REPLACEMENT" )
	AddConVarSetting( "ns_disallowed_weapons", "#DISALLOWED_WEAPONS" )
	AddConVarSetting( "ns_disallowed_weapon_primary_replacement", "#REPLACEMENT_WEAPON" )
	AddConVarSettingEnum( "ns_should_return_to_lobby", "#SHOULD_RETURN_TO_LOBBY", [ "#NO", "#YES" ] )
```

Reply:
should these even be settings? they aren't FCVAR_ARCHIVE so they won't save